src/main/java/com/tstepnik/planner/repository/TaskRepository.java

```diff
@@ -15,4 +16,10 @@
 
     Optional<Task> findById(Long id);
 
+    @Query("SELECT COUNT(t) FROM Task t WHERE t.userId = ?1 AND t.plannedFor <= ?2")
+    Long countAllUserArchivedTasks(Long userId, ZonedDateTime dateTime);
+
+    @Query("SELECT COUNT(t) FROM Task t WHERE t.done=true AND t.userId = ?1 AND t.plannedFor <= ?2")
+    Long countAllUserFinishAndArchivedTasks(Long userId, ZonedDateTime dateTime);
```

Comment:
Instead of using this method/query what are our other options of retrieving data we need?

Reply:
I don't know other options.

Reply:
There is something called Criteria in spring data. You should check it out :) Not saying we need to use it now, but good to know something like this exists